src/packages/log-viewer/workspace/views/overview/components/log-viewer-saved-searches-overview.element.ts

```diff
@@ -19,21 +32,54 @@ export class UmbLogViewerSavedSearchesOverviewElement extends UmbLitElement {
 			this.#logViewerContext?.getSavedSearches();
 			this.#observeStuff();
 		});
+		this.inputQuery$
+			.pipe(
+				tap(() => (this._showLoader = true)),
+				debounceTime(250)
+			)
+			.subscribe((query) => {
+				this.#logViewerContext?.setFilterExpression(query);
+				this.#persist(query);
+				this._isQuerySaved = this._savedSearches.some((search) => search.query === query);
+				this._showLoader = false;
+			});
 	}
 
 	#observeStuff() {
 		if (!this.#logViewerContext) return;
 		this.observe(this.#logViewerContext.savedSearches, (savedSearches) => {
 			this._savedSearches = savedSearches ?? [];
+			this._isQuerySaved = this._savedSearches.some((search) => search.query === this._inputQuery);
 		});
+
+		this.observe(this.#logViewerContext.filterExpression, (query) => {
+			this._inputQuery = query;
+			this._isQuerySaved = this._savedSearches.some((search) => search.query === query);
+		});
+	}
+
+	#persist(filter: string) {
+		let query = getQuery();
+
+		query = {
+			...query,
+			lq: filter,
+		};
+		const newPath = path().replace('/overview', '/search');
+
+		window.history.pushState({}, '', `${newPath}?${toQueryString(query)}`);
+	}
+
+	#setQueryFromSavedSearch(query: string) {
+		this.inputQuery$.next(query);
 	}
 
 	#renderSearchItem = (searchListItem: SavedLogSearchResponseModel) => {
 		return html` <li>
 			<uui-menu-item
 				label="${searchListItem.name ?? ''}"
 				title="${searchListItem.name ?? ''}"
-				href=${`section/settings/workspace/logviewer/search/?lq=${encodeURIComponent(searchListItem.query ?? '')}`}>
+				@click=${() => this.#setQueryFromSavedSearch(searchListItem.query ?? '')}>
```

Comment:
I'm a bit sorry to see this change from an actual link to an onclick, meaning you can no longer copy the link and use it elsewhere. The search page you land on after clicking the link should be able to handle the "lq" parameter simply if it detects it in the URL.

Reply:
What do you exactly mean by this? Do I need to keep `href` instead of `@click` attribute?

Reply:
Ah, got it, this issue was just related to a minor change in the `href` link, I unnecessarily made these changes, thought that it wasn't persisting.